chall12/szavarce.c

```diff
@@ -0,0 +1,102 @@
+#include <string.h>
+#include <stdlib.h>
+
+void	reverse_text(char *str, int start, int end)
+{
+	int		pos;
+	int		pos2;
+	char	tmp[end - start + 1];
+
+	pos = 0;
+	pos2 = 0;
+	while (pos < end - start + 1)
+	{
+		tmp[pos] = str[start + pos];
+		if (tmp[pos] == '(')
+			tmp[pos] = ')';
+		else if (tmp[pos] == ')')
+			tmp[pos] = '(';
+		pos++;
+	}
+	pos2 = pos - 1;
+	pos = 0;
+	while (pos < end - start + 1)
+	{
+		str[start + pos] = tmp[pos2--];
+		pos++;
+	}
+}
+
+int		check_first_parenthesis(const char *str)
+{
+	int		pos;
+
+	pos = 0;
+	while (str[pos] && str[pos] != '\0')
+	{
+		if (str[pos] == '(')
+			return (1);
+		if (str[pos] == ')')
+			return (0);
+		pos++;
+	}
+	return (1);
+}
+
+int		count_chars(const char *str, char to_find)
+{
+	int		pos;
+	int		count;
+
+	pos = 0;
+	count = 0;
+	while (str[pos] && str[pos] != '\0')
+	{
+		if (str[pos] == to_find)
+			count++;
+		pos++;
+	}
+	return (count);
+}
+
+char *ft_reverse_parenthesis(const char *str)
+{
+    int		pos;
+    int		count;
+	char	*new_str;
+	int		start_pos;
+	int		end_pos;
+
+	start_pos = 0;
+	end_pos = 0;
+	if (!(str) || (count_chars(str, '(') != count_chars(str, ')')))
+		return (0);
+	if (check_first_parenthesis(str) == 0)
+		return (0);
+	if (!(new_str = malloc(sizeof(char) * strlen(str))))
```

Comment:
You are not allocating the space for the null terminator of the string (`\0`) this is causing a segfault...